java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToSwitchPatternInstanceOf.java

```diff
@@ -155,12 +157,105 @@ protected void performRewrite(JavaFix.TransformationContext ctx) {
+        while (ifPath != null && ifPath.getKind() == Tree.Kind.IF) {
+            matchVarIndex++;
+            IfTree it = (IfTree) ifPath;
+            if (MatcherUtilities.matches(ctx, new TreePath(ctx.getPath(), it.getCondition()), "($expr" + matchVarIndex + " instanceof $typeI" + matchVarIndex + " $var" + matchVarIndex + ")", true)
+                    && MatcherUtilities.matches(ctx, new TreePath(ctx.getPath(), it.getThenStatement()), "{ $other" + matchVarIndex + "$;}", true)) {
+                if (ctx.getMultiVariables().get("$other" + matchVarIndex + "$").isEmpty()) {
+                    return null;
+                }
+                if (!(ctx.getVariables().get("$expr" + matchVarIndex).getLeaf().toString().equals(expr0))) {
+                    return null;
+                }
+                for (TreePath tp : ctx.getMultiVariables().get("$other" + matchVarIndex + "$")) {
+                    if (tp.getLeaf().getKind() == Tree.Kind.BREAK || tp.getLeaf().getKind() == Tree.Kind.CONTINUE) {
+                        return null;
+                    }
+                }
+            } else {
+                return null;
+            }
+            ifPath = it.getElseStatement();
+        }
+
+        Fix fix = new FixPatternMatchToSwitch(ctx.getInfo(), ctx.getPath(), false, Collections.emptySet()).toEditorFix();
+        return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_ConvertToSwitchPatternInstanceOf(), fix);
+
+    }
+
+    private static final class FixPatternMatchToSwitch extends JavaFix {
+        private final boolean removeFirst;
+
+        public FixPatternMatchToSwitch(CompilationInfo info, TreePath main,  boolean removeFirst, Set<TreePath> replaceOccurrences) {
+            super(info, main);
+            this.removeFirst = removeFirst;
+        }
+
+        @Override
+        protected String getText() {
+            return Bundle.FIX_ConvertToSwitchPatternInstanceOf();
+        }
 
-                Tree defaultTree = null;
-                defaultTree = blockTree.getStatements().size() == 1 && isValidCaseTree(blockTree.getStatements().get(0))? blockTree.getStatements().get(0) : thenBlock;
+        @Override
+        protected void performRewrite(JavaFix.TransformationContext ctx) {
+            WorkingCopy wc = ctx.getWorkingCopy();
+            TreePath main = ctx.getPath();
+            List<CaseTree> ctl = new LinkedList<>();
+            InstanceOfTree iot = null;
 
-                CaseTree casePatterns = wc.getTreeMaker().CasePatterns(caseBindPattern, defaultTree);
-                ctl.add(casePatterns);
+            Tree ifPath = ctx.getPath().getLeaf();
+            int matchVarIndex = 1;
+            List<IfTree> ifTrees = new ArrayList<>();
+            while (ifPath != null && ifPath.getKind() == Tree.Kind.IF) {
+                matchVarIndex++;
+                IfTree it = (IfTree) ifPath;
+                ifTrees.add(it);
+                ifPath = it.getElseStatement();
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 241:246, 144:149.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.